eawx-build/Native/FileLinkerFactory.cs

```diff
@@ -0,0 +1,19 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace EawXBuild.Native {
+    public class FileLinkerFactory : IFileLinkerFactory {
+        public IFileLinker MakeFileLinker() {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+                return new MacOSFileLinker();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+                return new WinFileLinker();
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+                return new LinuxFileLinker();
+
+            throw new InvalidOperationException();
```

Comment:
Optionally you could provide a reason like the OS being unsupported due to differences in the file system here - because technically someone got the thing to run only to fail now.